geniza/corpus/views.py

```diff
@@ -364,6 +363,41 @@ def get_context_data(self, **kwargs):
         return context_data
 
 
+class RelatedDocumentView(DocumentDetailView):
+    """List of :class:`~geniza.corpus.models.Document`
+    objects that are related to specific :class:`~geniza.corpus.models.Document`
+    (e.g., by occuring on the same shelfmark)."""
+
+    template_name = "corpus/related_documents.html"
+    viewname = "corpus:related-documents"
+
+    def page_title(self):
+        # Translators: title of related documents page
+        return _("Related documents for %(doc)s") % {"doc": self.get_object().title}
+
+    def page_description(self):
+        doc = self.get_object()
+        count = doc.related_documents.count()
+        # Translators: description of related documents page, for search engines
+        return ngettext(
+            "%(count)d related document",
+            "%(count)d related documents",
+            count,
+        ) % {
+            "count": count,
+        }
+
+    def get_context_data(self, **kwargs):
+        doc = self.get_object()
+        # if there are no related documents, don't serve out this page
+        if not doc.related_documents.count():
+            raise Http404
+
+        context_data = super().get_context_data(**kwargs)
+        context_data.update({"highlighting": {}})  # no solr highlighting
```

Comment:
Quick question, would this `context_data` ever be expected to have highlighting? Good as a failsafe anyway since the template will respond to highlighting, but just wanted to make sure I'm not missing somewhere highlighting would be added.

Reply:
No, there won't ever be highlighting because there's no keyword search available on this page.

I couldn't figure out a way to make the template not error without setting an empty highlighting here in the context. If you see an easy way to do this in the template, that would be nice.

Reply:
Would it be ok to do one of these two things in `dict_item`? And if so, which is preferable?

```py
    return dictionary.get(key, None) if isinstance(dictionary, dict) else None
```

```py
    try:
        return dictionary.get(key, None)
    except AttributeError:
        return None
```

Reply:
you're right, I think template tags & filters are supposed to fail silently

second option is preferable